File: I've Been Everywhere, Man/everywhere.php
<?php
/**
 * Problem: I've Been Everywhere, Man
 * https://open.kattis.com/problems/everywhere
 *
 * @author   Martin Björling
 * @email    martinbjorling@gmail.com
 * @version  1.0
 * @since    2015-11-25
 */

$answers = [];

$numOfTestCases = stream_get_line(STDIN, 1024, PHP_EOL);

for ($i = 0; $i < $numOfTestCases; $i++) {
	$numOfWorkTrips = stream_get_line(STDIN, 1024, PHP_EOL);

	$cities = [];
 
	for ($j = 0; $j < $numOfWorkTrips; $j++) {
		$nextCity = stream_get_line(STDIN, 1024, PHP_EOL);

		if (!in_array($nextCity, $cities)) {
			$cities[] = $nextCity;
		}
	}

 	$answers[] = count($cities) . "\n";
}

foreach ($answers as $value) {
	fwrite(STDOUT, $value);
}


?>

File: I've Been Everywhere, Man/Everywhere.java
/**
 * Problem: I've Been Everywhere, Man
 * https://open.kattis.com/problems/everywhere
 *
 * @author   Martin Björling
 * @email    martinbjorling@gmail.com
 * @version  1.0
 * @since    2015-11-25
 */

import java.lang.Math;
import java.util.Scanner;
import java.util.ArrayList;

public class Everywhere {
	
	public static void main(String[] args) {

		Scanner sc = new Scanner(System.in);

		int numOfTestCases = Integer.parseInt(sc.nextLine());

		ArrayList<String> answers = new ArrayList<String>();

		for (int i = 0; i < numOfTestCases; i++) {

			int numOfWorkTrips = Integer.parseInt(sc.nextLine());

			ArrayList<String> cities = new ArrayList<String>();

			for (int j = 0; j < numOfWorkTrips; j++) {
				String nextCity = sc.nextLine();

				if (!cities.contains(nextCity)) cities.add(nextCity);
			}

			answers.add(Integer.toString(cities.size()));
		}

		for (int i = 0; i < numOfTestCases; i++) {
			System.out.println(answers.get(i));
		}
	}
}